### DISCUSSION TITLE
Webhook should be registered automatically?

### DISCUSSION THREAD

**Discussion by XBeg9** [actions/actions-runner-controller#574] [Questions] [answered]
I have deployed the webhook and the operator is working without throwing any errors, but the webhook by itself doesn't appear inside the repository, now I am struggling if it should be and what it should look like? :) Would be great to have a bit more documentation around this.

Thanks

**Comment by jgrimaud-hum** [comment]
Loosely related, but I am curious as to why the HRA webhook is in controllers and not the api. Ultimately I am in the process of updating the source code on my local to use kubebuilder v3 and while finishing up the webhook scaffolding, I have noticed that kubebuilder doesn't appear to be used to create the HRA kind in the actions group, or to create the HRA webhook. It's interesting since they appear to have some of the scaffolding in place, but there is no record of them in the PROJECT file. 

I have no intention of hijacking the discussion, but I am curious about the design decisions and implementation. 

I can create a new discussion if OP would prefer

**Reply by mumoshu** [reply]
@jgrimaud-hum Maybe you're talking about some other webhook thing, like k8s admission webhooks? To me @XBeg9 seems to be talking about our webhook-based autoscaler's webhook server, which has nothing to do with k8s admission webhooks that kubebuilder can scaffold.

**Comment by mumoshu** [comment, marked as answer]
> Webhook should be registered automatically?

@XBeg9 Hey! Are you saying actions-runner-controller should register a webhook endpoint for you? I think that's out of the scope of a project like this :)

Anyway, you'd need to follow k8s's standard way to expose pods to the internet. You usually have to use NodePort service and a managed loadbalancer, or a k8s LoadBalancer service, or an ingress controller to expose the webhook-based autoscaler's webhook endpoint.

The webhook endpoint URL depends the solution you use so actions-runner-controller can do nothing for that.

Also beware to configure your firewall (for AWS that m right be a security groups ingress) appropriately. Once everything is configured, you can see if github is successfully sending webhook events by browsing the webhook configuration page on GitHub.

**Reply by mumoshu** [reply]
If you're using our helm chart, this is the relevant part of values:

https://github.com/actions-runner-controller/actions-runner-controller/blob/master/charts/actions-runner-controller/values.yaml#L148-L155

**Reply by XBeg9** [reply]
@mumoshu yeah, my question was more about if it was by design that it's not registering the webhook (btw, it does for runner actions, that's why I got confused). 
I am using a helm chart and made a LoadBalancer type for the webhook part. Instead of digging inside the code, the documentation was missing the part how the URL should look like and where should I put it.

Anyway, thanks for a great project.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
